projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/auth_config.py

```diff
@@ -0,0 +1,157 @@
+        using the same Jenkins machine
+
+        Also, another use-case is if you want to connect to two different Control Services with different authentication.
+        For example, during development, a Demo Control Service works with staging provider but in the
+        deployment in production works with Production. So in order to switch you need to re-login every time.
+        But now you can just switch tabs (with different environment variables set)
+        :return:
+        """
+        return os.getenv("VDK_BASE_CONFIG_FOLDER", str(Path.home()))
+
+
+class InMemAuthConfiguration:
+    """
+    A class providing the equivalent of an in-memory AuthConfigFolder.
+    This configuration is available as long as the python process is running,
+    and is lost when the process is terminated.
+    """
+
+    CREDS_KEY = "credentials"
+
+    def __init__(self):
+        self._config = dict()
+
+    def save_credentials(self, content: str) -> None:
+        self._config[InMemAuthConfiguration.CREDS_KEY] = content
+
+    def delete_credentials(self) -> None:
+        self._config[InMemAuthConfiguration.CREDS_KEY] = ""
+
+    def read_credentials(self) -> Optional[str]:
+        return self._config.get(InMemAuthConfiguration.CREDS_KEY)
+
+
+class AuthConfigFolder:
+    """
+    A class responsible for managing the configuration files in the vdk configuration
+    folder
+    """
+
+    # TODO: this looks robust: https://github.com/ActiveState/appdirs
+    CONFIG_FOLDER_NAME = ".vdk.internal"
+    CREDENTIALS_FILE = "vdk-cred.json"
+    CONFIGURATION_FILE = "vdk-configuration.ini"
+
+    def __init__(self, base_dir=AuthConfig().local_config_folder):
+        self.vdk_config_folder = os.path.join(
+            base_dir, AuthConfigFolder.CONFIG_FOLDER_NAME
+        )
+        if os.path.isfile(self.vdk_config_folder):
+            raise VDKAuthException(
+                what="Credentials file was not created",
+                why=f"There is a file named {self.vdk_config_folder}",
+                consequence="User won't be able to access Control Service",
+                countermeasure=f"Remove file {self.vdk_config_folder}",
+            )
+        else:
+            if not os.path.exists(self.vdk_config_folder) and not os.path.isdir(
+                self.vdk_config_folder
+            ):
+                try:
+                    pathlib.Path(self.vdk_config_folder).mkdir(
+                        parents=True, exist_ok=True
+                    )
+                except Exception as e:
+                    raise VDKAuthException(
+                        what="Configuration folder was not created and user is not logged in",
+                        why=f"Cannot create: {self.vdk_config_folder} configuration folder: {str(e)}",
+                        consequence="User won't be able to access Control Service",
+                        countermeasure=f"Check if the client has write access to: {base_dir}",
+                    )
```

Comment:
```suggestion
                    raise VDKAuthException(
                        what="Configuration folder was not created and user is not logged in",
                        why=f"Cannot create: {self.vdk_config_folder} configuration folder: {str(e)}",
                        consequence="User won't be able to access Control Service",
                        countermeasure=f"Check if the client has write access to: {base_dir}",
                    ) from e
```
Is including the exception string inside the why parameter just as good as raising from the original exception? It might be, and if so you can ignore this suggestion.

Reply:
I am hesitant to excessively clutter the logs, but if others think the whole exception should be logged, as well, I am fine to go with the suggested change.